packages/cli/config/jestProfile.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const fs = require('fs-extra');
+const v8Profiler = require('v8-profiler-next');
+
+v8Profiler.setGenerateType(1);
+
+beforeEach(() => {
+  const title = expect.getState().currentTestName;
+  v8Profiler.startProfiling(title, true);
+  v8Profiler.startSamplingHeapProfiling();
+  fs.mkdirSync('./profiles', { recursive: true });
```

Comment:
Are you expecting something to delete this directory? If not, could this be done once in the `beforeAll`.

Reply:
This is just a draft PR, not to be even merged. Just wanted to try this out and share if someone is interested.